simulator/func_sim/alu.h

```diff
@@ -236,6 +236,18 @@ struct ALU
 
     // Bit manipulations
     template<typename I> static void sbext( I* instr) { instr->v_dst[0] = 1U & ( instr->v_src[0] >> shamt_v_src2<typename I::RegisterUInt>( instr)); }
+    
+    template<typename I> static 
+    void max( I* instr)
+    { 
+        instr->v_dst[0] = narrow_cast<typename I::RegisterSInt>( instr->v_src[0]) >= narrow_cast<typename I::RegisterSInt>( instr->v_src[1]) ? instr->v_src[0] : instr->v_src[1];
```

Comment:
I think we have that in our predicate macros:

https://github.com/MIPT-ILab/mipt-mips/blob/f75708e2b1b683cfb21ed2029c17a212a0fac25c/simulator/func_sim/alu.h#L127-L128

so may be even less verbose:

```c++
instr->v_dst[0] = instr->v_src[ge( instr) ? 0 : 1];
```

Reply:
Corrected and tested it.